aws-xray-recorder-sdk-sql/src/main/java/com/amazonaws/xray/sql/TracingStatement.java

```diff
@@ -0,0 +1,134 @@
+                    openDummySegment = true;
+                    System.out.println(sanitizeSql(resolveExecutedSql(method.getName(), args)));
+                    AWSXRay.beginDummySegment();
+                }
+                Subsegment subsegment = AWSXRay.beginSubsegment("SQL");
+                subsegment.putAllSql(extractSqlParams(method, args));
+                subsegment.setNamespace(Namespace.REMOTE.toString());
+                try {
+                    return method.invoke(original, args); //execute the query
+                } catch (Throwable t) {
+                    subsegment.addException(t);
+                    if (t instanceof InvocationTargetException && t.getCause() != null) {
+                        throw t.getCause();
+                    } else {
+                        throw t;
+                    }
+                } finally {
+                    AWSXRay.endSubsegment();
+                    if (openDummySegment) {
+                        AWSXRay.endSegment();
+                    }
+                }
+
+            } else {
+                //else, simply delegates
+                return method.invoke(original, args);
+            }
+        }
+
+        private Map<String, Object> extractSqlParams(Method method, Object[] args) {
+            Map<String, Object> additionalParams = new HashMap<>();
+            try {
+                DatabaseMetaData metadata = original.getConnection().getMetaData();
+                additionalParams.put("url", metadata.getURL());
+                additionalParams.put("driver_version", metadata.getDriverVersion());
+                additionalParams.put("database_type", metadata.getDatabaseProductName());
+                additionalParams.put("database_version", metadata.getDatabaseProductVersion());
+                additionalParams.put("sanitized_query", sanitizeSql(resolveExecutedSql(method.getName(), args)));
+            } catch (SQLException e) {
+                //a SqlException at this stage is kind of strange ... just ignore
+                // (i.e the XRay trace will not contain the info but this is not very severe)
+                //at least, we provide a way to identify the problem in the XRay trace
+                additionalParams.put("url", "strange SQLException occurred ...");
+            }
+            return additionalParams;
+        }
+
+        private String resolveExecutedSql(String invokedMethodName, Object[] invokedMethodArgs) {
+            switch (invokedMethodName) {
+                case "execute":
+                case "executeQuery":
+                case "executeUpdate":
+                    return invokedMethodArgs != null && invokedMethodArgs.length > 0 && (invokedMethodArgs[0] instanceof String) ? (String) invokedMethodArgs[0] : sql;
+                case "executeBatch":
+                    //retrieving the executed queries is possible ... but that would produce a very big output !
+                    return "BATCH";
```

Comment:
I would suggest truncation because BATCH can be confusing as itself is possibly a SQL input